pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConnectionPool.java

```diff
@@ -170,6 +189,15 @@ public void initChannel(SocketChannel ch) throws Exception {
                 cnx.ctx().close();
                 return null;
             });
+
+            // close the connection after connection-lifetime completes
+            if (connectionLifetimeInSecond > 0) {
+                scheduler.schedule(() -> {
+                    if (cnx != null && cnx.channel().isActive()) {
+                        cnx.channel().disconnect();
+                    }
+                }, connectionLifetimeInSecond, TimeUnit.SECONDS);
+            }
```

Comment:
Are we closing the connection irrespective of it's active or not? We should close it only if it's inactive for the last n minutes.